quadriga/src/main/java/edu/asu/diging/quadriga/api/v1/AddNetworkApiController.java

```diff
@@ -59,31 +59,43 @@
     @ResponseBody
     @RequestMapping(value = "/api/v1/collection/{collectionId}/network/add", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
     public HttpStatus processJson(@RequestBody Quadruple quadruple, @PathVariable String collectionId) {
+
+        // First we check whether a quadruple is present in request body
+        if (quadruple == null) {
+            logger.error("Quadruple not present in network submission request for collectionId: "  + collectionId);
+            return HttpStatus.NO_CONTENT;
+        }
         
+        // Next, we check whether a collection and mappedTripleGroup is present
         // Every time a new network is submitted, the triple in that network has to be added as the
         // default MappedTripleGroup for given collectionId
         MappedTripleGroup mappedTripleGroup;
         try {
             mappedTripleGroup = mappedTripleGroupService.get(collectionId, MappedTripleType.DEFAULT_MAPPING);
             if(mappedTripleGroup == null) {
-                logger.error("Couldn't find or persist a new MappedTripleGroup entry for collectionId: " + collectionId);
                 return HttpStatus.NOT_FOUND;
             }
         } catch(InvalidObjectIdException | CollectionNotFoundException e)  {
-            logger.error("Couldn't submit network", e);
+            logger.error(e.getMessage());
             return HttpStatus.NOT_FOUND;
         }
 
-        if (quadruple == null) {
-            return HttpStatus.NO_CONTENT;
-        }
-
         // save network
         List<CreationEvent> events = networkMapper.mapNetworkToEvents(quadruple.getGraph());
         List<EventGraph> eventGraphs = events.stream().map(e -> new EventGraph(e)).collect(Collectors.toList());
         eventGraphs.forEach(e -> {
             e.setCollectionId(new ObjectId(collectionId));
             e.setDefaultMapping(quadruple.getGraph().getMetadata().getDefaultMapping());
+            e.setContext(quadruple.getGraph().getMetadata().getContext());
+
+            /**
+             * A new story will later be created to get info about just one app from citesphere using OAuth token.
+             * This app's name should be stored in eventGraph instead of the client id
+             * Until that story is done, we need to store clientId instead of appName
+             * We can't store clientId yet as it depends on story Q20-3
+             * After merging story Q20-3, this needs to be changed to tokenInfo.getClientId()
+             */
+            e.setSubmittingApp("AppName");
```

Comment:
There should be a `FIXME` in the comment

Reply:
Adding a FIXME raises an issue in codefactor check.